File: 프로그래머스/lv2/12900. 2 x n 타일링/2 x n 타일링.py
def solution(n):
    answer = 0
    
    num1 = 1
    num2 = 2
    for i in range(1,n-1):
        num2 = (num1 + num2)%1000000007
        num1 = num2 - num1
            
    
    return num2

File: 프로그래머스/2/12900. 2 x n 타일링/2 x n 타일링.java
class Solution {
    public int solution(int n) {
        int answer = 0;
        
        int[] dp = new int[n+1];
        
        dp[1] = 1 % 1000000007;
        dp[2] = 2 % 1000000007;
        
        for (int i = 3; i<dp.length;i++) {
            dp[i] = (dp[i-1] + dp[i-2]) % 1000000007;
        }
        return dp[n];
        
    }
}